### PR TITLE
Transition Razor test comparers to new project.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Changed namespaces for test comparers.
- Did project.json format change in separate commit.

#461

### CODE REVIEW

File: test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperDesignTimeDescriptorComparer.cs [hunk 1]
```diff
@@ -1,41 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using Microsoft.Internal.Web.Utils;
-
-namespace Microsoft.AspNet.Razor.TagHelpers
-{
-    public class TagHelperDesignTimeDescriptorComparer : IEqualityComparer<TagHelperDesignTimeDescriptor>
```
Comment (dougbu) [inline test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperDesignTimeDescriptorComparer.cs]: Nice to get rid of this duplicate class.

File: src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs [hunk 2]
```diff
@@ -6,9 +6,9 @@
 using Microsoft.AspNet.Razor.TagHelpers;
 using Microsoft.Internal.Web.Utils;
 
-namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+namespace Microsoft.AspNet.Razor.Test.Internal
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs]: Shouldn't have `.Test` in the namespace except for test infrastructure classes. So `.Internal` seems redundant. Why not just `Microsoft.AspNet.Razor.Test`?

File: src/Microsoft.AspNet.Razor.Test.Sources/project.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿{
+  "version": "1.0.0-*",
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor.Test.Sources/project.json]: Odd for this project to be inconsistent with the rest of the repo. Suggest `4.0.0-*`

**Comment by dougbu** [comment]
In the aspnet/Common repo, we've required a test class for every shared class. Probably useful to do that here as well.

**Comment by dougbu** [comment]
:watch:

File: src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs (hunk 2 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Test.Sources/CaseSensitiveTagHelperDescriptorComparer.cs]: `.Internal` is consistent with `aspnet/Common`

File: src/Microsoft.AspNet.Razor.Test.Sources/project.json (hunk 3 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Test.Sources/project.json]: I thought of that as well, but this isn't specific to the other Razor dependencies. Don't feel strongly about it though. /cc @Eilon

**Comment by NTaylorMullen** [comment]
@dougbu these are internal utilities for tests. We don't typically test test utilities.

File: src/Microsoft.AspNet.Razor.Test.Sources/project.json (hunk 3 above)
Comment (Eilon) [inline src/Microsoft.AspNet.Razor.Test.Sources/project.json]: Meh don't care :smile:

**Comment by Eilon** [comment]
This is just shuffling code around, so :shipit:

**Pull request merged**